Code/Framework/AzCore/Tests/DOM/DomPathTests.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/DOM/DomPath.h>
+#include <AzCore/Name/NameDictionary.h>
+#include <Tests/DOM/DomFixtures.h>
+
+namespace AZ::Dom::Tests
+{
+    class DomPathTests : public DomTestFixture
+    {
+    };
+
+    TEST_F(DomPathTests, EmptyPath_IsEmpty)
+    {
+        Path path("");
+        EXPECT_EQ(path.GetEntries().size(), 0);
+    }
+
+    TEST_F(DomPathTests, EmptyPath_IsEqualToDefault)
+    {
+        EXPECT_EQ(Path(""), Path());
+    }
+
+    TEST_F(DomPathTests, Index_IsNumeric)
+    {
+        EXPECT_EQ(Path("/0")[0], 0);
+        EXPECT_EQ(Path("/20")[0], 20);
+        EXPECT_EQ(Path("/9999")[0], 9999);
+        EXPECT_EQ(Path("/0/4/5/1")[3], 1);
+    }
+
+    TEST_F(DomPathTests, Index_ConvertsToString)
+    {
+        Path p;
+        EXPECT_EQ(p.ToString(), "");
+
+        p.Push(0);
+        EXPECT_EQ(p.ToString(), "/0");
+
+        p.Push(1);
+        EXPECT_EQ(p.ToString(), "/0/1");
+
+        p.Push(10);
+        EXPECT_EQ(p.ToString(), "/0/1/10");
+
+        p.Push(9999);
+        EXPECT_EQ(p.ToString(), "/0/1/10/9999");
+
+        p.Pop();
+        EXPECT_EQ(p.ToString(), "/0/1/10");
+    }
+
+    TEST_F(DomPathTests, Key_IsString)
+    {
+        EXPECT_EQ(Path("/foo")[0], "foo");
+        EXPECT_EQ(Path("/bar")[0], "bar");
+        EXPECT_EQ(Path("/foo/bar/baz12345")[0], "foo");
+        EXPECT_EQ(Path("/foo/bar/baz12345")[2], "baz12345");
+        EXPECT_EQ(Path("//foo")[0], "");
```

Comment:
question: Does that mean paths like "//foo////bar" are valid Dom paths?

Reply:
They are, each empty segment represents an object entry with a value of `""` This is pretty esoteric, so I wouldn't consider it a common case, just a technically valid one.

Reply:
That should be valid if the [JSON Pointer spec](https://datatracker.ietf.org/doc/html/rfc6901) is followed.
`/` is not a root separator, it is part of each individual "path segment".
>
 The ABNF syntax of a JSON Pointer is:

      json-pointer    = *( "/" reference-token )
      reference-token = *( unescaped / escaped )
      unescaped       = %x00-2E / %x30-7D / %x7F-10FFFF
         ; %x2F ('/') and %x7E ('~') are excluded from 'unescaped'
      escaped         = "~" ( "0" / "1" )
        ; representing '~' and '/', respectively

I point out the emphasis on the reference_token rule: "*(unescaped/ escaped)", where it accepts 0 or more tokens.

So "/" represents an a member with a key value of ""

Reply:
For example the following is valid JSON

```JSON
{
   "" : {
       "foo": {
          "": {
             "": {
               "": {
                   "bar": true
               }
            }  
         }
       }
   }
}
```

That boolean true would be represented with the path of "//foo////bar"